### PR TITLE
Updates the API within the core project.

### PR DESCRIPTION
**Pull request by otaviojava**
## These updates include:

* Update to foreach
* Unnecessary unboxing
* Simplifies code with the newest method at Map
* Simplifies code with lambda expressions

This PR does not have performance progress, but it uses new API to make the code more readable and easy to maintain.

### CODE REVIEW

**Comment by spmallette** [comment]
sorry - i've been in the midst of releasing TinkerPop so i haven't had a chance to take a look at this yet. i can probably get to this by the end of the week.

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/AbstractThreadLocalTransaction.java [hunk 1]
```diff
@@ -38,25 +38,12 @@
...
-        };
+            ThreadLocal.withInitial(() -> CLOSE_BEHAVIOR.ROLLBACK);
     
-    protected final ThreadLocal<List<Consumer<Transaction.Status>>> transactionListeners = new ThreadLocal<List<Consumer<Transaction.Status>>>() {
-        @Override
-        protected List<Consumer<Transaction.Status>> initialValue() {
-            return new ArrayList<>();
-        }
-    };
+    protected final ThreadLocal<List<Consumer<Transaction.Status>>> transactionListeners = ThreadLocal.withInitial(() -> new ArrayList<>());
```
[the remark this answers is not in the document]
Comment (otaviojava) [reply gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/util/AbstractThreadLocalTransaction.java]: Thank you, fixed.

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/decoration/EventStrategy.java [hunk 2]
```diff
@@ -223,11 +223,7 @@ public void addEvent(final Event evt) {
      */
     public static class TransactionalEventQueue implements EventQueue {
 
-        private final ThreadLocal<Deque<Event>> eventQueue = new ThreadLocal<Deque<Event>>() {
-            protected Deque<Event> initialValue() {
-                return new ArrayDeque<>();
-            }
-        };
+        private final ThreadLocal<Deque<Event>> eventQueue = ThreadLocal.withInitial(() -> new ArrayDeque<>());
```
[the remark this answers is not in the document]
Comment (otaviojava) [reply gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/decoration/EventStrategy.java]: Thank you, fixed.

**Comment by otaviojava** [comment]
retest this please

**Comment by FlorianHockmann** [comment]
> retest this please

I just triggered a new build on Travis.

**Comment by spmallette** [comment]
If we have to make these sorts of changes, I think that this PR should target the `tp33` branch so that both of our release branches have the same consistent look to them.

**Comment by otaviojava** [comment]
Nice, as soon it is merged we can work in a cherry-pick :)

**Comment by spmallette** [comment]
I hate to be a bother but our git workflow goes from `tp33 --> master`. we typically don't make it a habit of cherrypicking commits backwards in the flow as it makes the git history messy. 

i'd much prefer a rebase on tp33 with a squash of all these commits into one. we can then merge forward to master. if you need to apply additional changes on master as they only apply there then that can come as a separate PR. In general, it's best to always target tp33 with your PRs unless there is explicit reason not to. If you ever have doubts on what branch to target, feel free to ask ahead of time. 
sorry for the hassle and thank you.

**Pull request closed without merging**